rusk/src/lib/chain/rusk.rs

```diff
@@ -125,11 +129,14 @@ impl Rusk {
                 Ok(receipt) => {
                     let gas_spent = receipt.gas_spent;
 
-                    // If the transaction went over the block gas limit we
-                    // re-execute all spent transactions. We don't discard the
-                    // transaction, since it is technically valid.
                     if gas_spent > block_gas_left {
-                        warn!("This is not supposed to happen with conservative tx inclusion");
+                        // If the transaction went over the block gas limit we
+                        // re-execute all spent transactions. We don't discard
+                        // the transaction, since it is
+                        // technically valid.
+                        if gas_spent > block_gas_left {
+                            warn!("This is not supposed to happen with conservative tx inclusion");
+                        }
```

Comment:
The nested `if` has the same clause is this on purpose? :confused:

Reply:
There was one more condition in line 132 which was later removed, I will correct that.